lib/pf/cmd/pf/import/nodes.pm

```diff
@@ -15,24 +15,93 @@ use strict;
 use warnings;
 use base qw(pf::cmd);
 use Role::Tiny::With;
+use pf::constants qw($TRUE $FALSE);
 use pf::constants::exit_code qw($EXIT_SUCCESS);
+use pf::file_paths qw($install_dir);
 with 'pf::cmd::roles::show_parent_help';
 
+=head2 parseArgs
+
+Parse the arguments for this command
+
+=cut
+
 sub parseArgs {
     my ($self) = @_;
-    my ($file) = $self->args;
-    if($file) {
-        $self->{file} = $file;
-        return 1;
+
+    my ( $file, @params ) = $self->args;
+
+    unless(defined($file)) {
+        print STDERR "You must specify a file name\n";
+        return $FALSE;
+    }
+
+    unless(-f $file) {
+        print STDERR "File $file doesn't exist...\n";
+        return $FALSE;
     }
-    return 0;
+
+    require pf::config;
+
+    my %params = (
+        'columns'           => "mac",
+        'default-role'      => "default",
+        'default-unregdate' => "2038-01-01 00:00:00",
+        'default-voip'      => "no",
+        'default-owner'     => "default",
+        'delimiter'         => "comma",
+    );
+    foreach my $param ( @params ) {
+        my @data = split('=', $param);
+        if ( exists($params{$data[0]}) ) {
+            if ( length($data[1]) >= 1 ) {
+                $params{$data[0]} = $data[1];
+            }
+            else {
+                print STDERR "Invalid parameter value '$data[1]' for parameter '$data[0]'\n";
+            }
+        }
+        else {
+            print STDERR "Unknown parameter '$data[0]'\n";
+        }
+    }
+
+    my @columns = map { { enabled => 1, name => $_ } } split(/\s*,\s*/, $params{columns});
+
+    $self->{params} = {
+        %params,
+        filename => $file,
+        columns => \@columns,
+    };
+
+    return $TRUE;
 }
 
+=head2 _run
+
+Run the import
+
+=cut
+
 sub _run {
     my ($self) = @_;
+    require pf::nodecategory;
     require pf::import;
-    pf::import::nodes($self->{file});
-    print "Import process complete\n";
+    
+    my $params = $self->{params};
+    my ($status, $info) = pf::import::nodes(
+        $params->{filename},
+        {
+            columns => $params->{columns},
+            delimiter => $params->{delimiter},
+            default_pid => $params->{"default-owner"},
+            default_category_id => pf::nodecategory::nodecategory_view_by_name($params->{"default-role"})->{category_id},
+            default_voip => $params->{"default-voip"},
+            default_unregdate => $params->{"default-unregdate"},
+        },
+    );
```

Comment:
No error checking of the status

Reply:
It prints to STDOUT the result of the import